CONTRIBUTING.rst

```diff
@@ -469,34 +520,112 @@ Type Hints
   ``x: JSON`` the expression ``x['a']['b']['c']`` would be of type ``JSON``
   while ``x['a']['b']['c']['d']`` would be of type ``Any``.
 
-  
+
+Method and function arguments
+-----------------------------
+
+* Arguments declared as a keyword must be passed as keyword arguments at all
+  call sites.
+
+* For call sites with more than three passed arguments, all arguments should be
+  passed as keywords, even positional arguments, if one of the arguments is
+  passed as a keyword.
+
+* At call sites that pass a literal expression to a function or method, consider
+  passing the argument as a keyword. Instead of ::
+
+    foo(x, {})
+    bar(True)
+
+  use ::
+
+    foo(filters={})
+    bar(delete=True)
+
+  while leaving ::
+
+    add(1, 2)
+    setDelete(True)
+
+  as is.
+
+* We prefer enforcing the use of keyword arguments using keyword-only arguments
+  as defined in `PEP-3102`_.
+
+
+.. _PEP-3102: https://www.python.org/dev/peps/pep-3102/
+
+
 Testing
 =======
 
-* All code should be covered by unit tests.
+Coverage of new code
+--------------------
+
+* All new code should be covered by unit tests.
+
+Coverage of legacy code
+-----------------------
 
 * Legacy code for which tests were never written should be covered when it is
   modified.
-  
+
+Subtests
+--------
+
 * Combinatorial tests (tests that exercise a number of combinations of inputs)
   should make use of ``unittest.TestCase.subTest()`` so a single failing
   combination doesn't prevent other combinations form being exercised.
 
+* Sub-tests may makes sense even when there isn't a large number of
+  combinations. Consider two independent tests that share an expensive fixture.
+  Instead of isolating the two tests in separate `TestCase` whose ``setUpClass``
```

Comment:
```suggestion
  Instead of isolating the two tests in separate ``TestCase`` whose ``setUpClass``
```

Reply:
I searched for these with a regex. So they should all be fixed.